LeapEngine/Leap.cpp

```diff
@@ -106,14 +122,18 @@ void leap::LeapEngine::Run(int desiredFPS)
         sceneManager.OnGUI();
         gameContext.OnGUI();
 
+        m_pRenderer->DrawTriangles(physics.GetDebugDrawings());
         m_pRenderer->Draw();
         glfwSwapBuffers(m_pWindow);
 
         sceneManager.OnFrameEnd();
 
-        //Sleep to sync back with desired fps
-        const auto sleepTimeMs = frameTimeMs - std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::high_resolution_clock::now() - currentTime).count();
-        std::this_thread::sleep_for(std::chrono::milliseconds(static_cast<int>(sleepTimeMs)));
+        // Wait to sync back with desired fps
+        long long curFrameTimeNs{};
+        do
+        {
+            curFrameTimeNs = (std::chrono::high_resolution_clock::now() - currentTime).count();
+        } while (frameTimeNs - curFrameTimeNs > 0);
```

Comment:
This will set the CPU thread on 100% for limited frames. Fixes it for now but should be looked into.